src/main/java/org/commonwl/view/workflow/WorkflowRESTController.java

```diff
@@ -88,14 +88,18 @@ public WorkflowRESTController(WorkflowFormValidator workflowFormValidator,
     /**
      * Create a new workflow from the given URL
      * @param url The URL of the workflow
+     * @param branch The branch where the workflow can be found
+     * @param path The path within the repository to the workflow file
      * @return Appropriate response code and optional JSON string with message
      */
     @PostMapping(value = "/workflows", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
-    public ResponseEntity<?> newWorkflowFromGithubURLJson(@RequestParam(value="url") String url,
-                                                          HttpServletResponse response) {
+    public ResponseEntity<?> newWorkflowFromGitURLJson(@RequestParam(value="url") String url,
+                                                       @RequestParam(value="branch", required=false) String branch,
+                                                       @RequestParam(value="path", required=false) String path,
```

Comment:
Which one will win if both `path` are provided and a "smart" `url` like https://github.com/common-workflow-language/workflows/blob/master/workflows/lobSTR/lobSTR-workflow.cwl ?